x-pack/plugins/serverless/server/plugin.ts

```diff
@@ -75,6 +76,8 @@ export class ServerlessPlugin implements Plugin<ServerlessPluginSetup, Serverles
       );
     }
 
+    core.uiSettings.setAllowlist(ALL_ALLOWLIST_SETTINGS);
```

Comment:
We need to expand this, and restrict `setAllowList` to be called one time.  We can also have `setAllowList` call `setReadOnlyMode('strict')`.

```
private setupProjectSettings(keys: string[] = []): void => {
  const settings = [...ALL_ALLOWLIST_SETTINGS].concat(keys);
  core.uiSettings.setAllowlist(settings);
}

// ...
public setup(core: CoreSetup) {
// ...
return {
  setupProjectSettings: this.setupProjectSettings
};
```
I can open a PR against this to demonstrate further, if necessary.

Reply:
Thanks for the example, I added the `setupProjectSettings` from the serverless plugin and called it from the serverless project plugins as you suggested.

How would you suggest approaching restricting `setAllowlist` to be called only once? I was thinking about setting a private variable to false and toggling it when `setAllowlist` is called, but I remember you mentioned something about setting `eslint` rules. Can you share more about this?